website/public/js/services/chatServices.js

```diff
@@ -1,33 +1,85 @@
-    {
-      postChat: {method: "POST", url: ApiUrl.get() + '/api/v2/groups/:gid/chat'},
-      like: {method: 'POST', isArray: true, url: ApiUrl.get() + '/api/v2/groups/:gid/chat/:messageId/like'},
-      deleteChatMessage: {method: "DELETE", url: ApiUrl.get() + '/api/v2/groups/:gid/chat/:messageId'},
-      flagChatMessage: {method: "POST", url: ApiUrl.get() + '/api/v2/groups/:gid/chat/:messageId/flag'},
-      clearFlagCount: {method: "POST", url: ApiUrl.get() + '/api/v2/groups/:gid/chat/:messageId/clearflags'},
-    });
-
-  var chatService = {
-    seenMessage: seenMessage,
-    clearCards: clearCards,
-    utils: utils
-  };
-
-  return chatService;
-
-  function clearCards() {
-    User.user.ops.update && User.set({'flags.cardReceived':false});
-  }
-
-  function seenMessage(gid) {
-    // On enter, set chat message to "seen"
-    $http.post(ApiUrl.get() + '/api/v2/groups/'+gid+'/chat/seen');
-    if (User.user.newMessages) delete User.user.newMessages[gid];
-  }
-}]);
+angular.module('habitrpg')
+.factory('Chat', ['$resource', '$http', 'ApiUrl', 'User',
+  function($resource, $http, ApiUrl, User) {
+    var apiV3Prefix = '/api/v3';
+
+    function getChat (groupId) {
+      return $http({
+        method: 'GET',
+        url: apiV3Prefix + '/groups/' + groupId + '/chat',
+      });
+    }
+
+    function postChat (groupId, message, previousMsg) {
+      var url = apiV3Prefix + '/groups/' + groupId + '/chat';
+
+      if (previousMsg) {
+        url += '?previousMsg=' + previousMsg;
+      }
+
+      return $http({
+        method: 'POST',
+        url: url,
+        data: {
+          message: message,
+        }
+      });
+    }
+
+    function deleteChat (groupId, chatId, previousMsg) {
+      var url = apiV3Prefix + '/groups/' + groupId + '/chat/' + chatId;
+
+      if (previousMsg) {
+        url += '?previousMsg=' + previousMsg;
+      }
+
+      return $http({
+        method: 'DELETE',
+        url: url,
+      });
+    }
+
+    function like (groupId, chatId) {
+      return $http({
+        method: 'POST',
+        url: apiV3Prefix + '/groups/' + groupId + '/chat/' + chatId + '/like',
+      });
+    }
+
+    function flagChatMessage (groupId, chatId) {
+      return $http({
+        method: 'POST',
+        url: apiV3Prefix + '/groups/' + groupId + '/chat/' + chatId + '/flag',
+      });
+    }
+
+    function clearFlagCount (groupId, chatId) {
+      return $http({
+        method: 'POST',
+        url: apiV3Prefix + '/groups/' + groupId + '/chat/' + chatId + '/clearflags',
+      });
+    }
+
+    function markChatSeen (groupId) {
+      return $http({
+        method: 'POST',
+        url: apiV3Prefix + '/groups/' + groupId + '/chat/seen',
+      });
+    }
+
+    return {
+      getChat: getChat,
+      postChat: postChat,
+      deleteChat: deleteChat,
+      like: like,
+      flagChatMessage: flagChatMessage,
+      clearFlagCount: clearFlagCount,
+      markChatSeen: markChatSeen,
+    }
+
+    function clearCards() {
```

Comment:
I think it's not exported because it's an User operation and they'll be ported later